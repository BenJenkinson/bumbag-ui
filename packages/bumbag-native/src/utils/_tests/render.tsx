import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
