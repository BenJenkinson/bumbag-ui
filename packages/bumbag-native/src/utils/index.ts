export * from './breakpoint';
export * from './getCSSFromStyleObject';
export * from './theme';
export * from './throttle';
export * from './useBreakpoint';
export * from './useStyle';
export * from './useTheme';
