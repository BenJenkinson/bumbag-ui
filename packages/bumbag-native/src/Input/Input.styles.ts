import { TextInput } from 'react-native';

import { AnimatedText } from '../Animated';
import { styled } from '../styled';
import { borderRadius, fontSize, palette, theme } from '../utils/theme';

export const StyledInput = styled(TextInput)`
  background: ${palette('default')};
  font-size: ${(props) => `${fontSize('200')(props)}px`};
  height: 44px;

  ${getVariantStyles}
  ${getDisabledStyles}

  ${theme('native.Input', 'styles.base')};
`;

function getVariantStyles(props) {
  if (props.variant === 'bordered') {
    return `
      border: 1px solid ${palette('white900', { dark: 'gray700' })(props)};
      border-radius: ${borderRadius('default')(props)};
      padding: 6.4px 12.8px;

      ${
        props.focus
          ? `
            border: 2px solid ${palette('primary')(props)};
            padding: 6.4px 11.8px;
          `
          : ''
      }
    `;
  }
  if (props.variant === 'underline') {
    return `
      border-bottom-width: 1px;
      border-bottom-color: ${palette('white900', { dark: 'gray700' })(props)};
      border-radius: 0px;
      padding: 6.4px 0px;

      ${
        props.focus
          ? `
            border-bottom-color: ${palette('primary')(props)};
            border-bottom-width: 2px;
          `
          : ''
      }
    `;
  }
  if (props.variant === 'borderless') {
    return `
      padding: 6.4px 0px;
    `;
  }
  return '';
}

function getDisabledStyles(props) {
  return props.disabled
    ? `
      background-color: ${palette('white700', { dark: 'black200' })(props)};
      color: ${palette('text100')(props)};
      cursor: not-allowed;

      ${theme('native.Input', `styles.disabled`)(props)};
    `
    : '';
}

//////////////////////////////////////////////////////////////////

export const StyledAnimatedLabel = styled(AnimatedText)`
  position: absolute;

  ${getAnimatedLabelVariantStyles}

  ${theme('native.Input', 'AnimatedLabel.styles.base')};
`;

function getAnimatedLabelVariantStyles(props) {
  if (props.variant === 'bordered') {
    return `
      background-color: white;
      margin: 0px 12.8px;
      padding: 0px 4px;
    `;
  }
  if (props.variant === 'underline') {
    return ``;
  }
  if (props.variant === 'borderless') {
    return ``;
  }
  return '';
}
