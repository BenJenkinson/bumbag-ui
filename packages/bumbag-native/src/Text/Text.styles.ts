import { Animated, Platform, Text } from 'react-native';
import { styled } from '../styled';
import { font, palette, theme } from '../utils/theme';

function getTextStyles(props) {
  return `
    color: ${palette('text')(props)};

    ${
      font('default')(props)
        ? `
            font-family: ${font('default')(props)};
          `
        : ''
    }

    ${
      Platform.OS === 'web'
        ? `
            font-size: unset;
          `
        : ''
    }

    ${theme('native.Text', 'styles.base')(props)};
  `;
}

export const StyledText = styled(Text)`
  ${getTextStyles}
`;

export const StyledAnimatedText = styled(Animated.Text)`
  ${getTextStyles}
`;
