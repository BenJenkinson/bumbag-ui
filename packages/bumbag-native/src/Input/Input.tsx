import * as React from 'react';
import { Platform, TextInputProps as RNTextInputProps } from 'react-native';
import { useFocus } from 'react-native-web-hooks';
import { Input as WebInput } from 'bumbag/Input';
import {
  createComponent,
  createElement,
  createHook,
  mergeRefs,
  omitCSSProps,
  pickCSSProps,
  useTheme,
} from 'bumbag/utils';

import { Animated } from '../Animated';
import { Box, BoxProps } from '../Box';
import { palette, usePalette } from '../utils';
import * as styles from './Input.styles';

export type LocalInputProps = {
  label?: string;
  labelProps?: any;
};
export type InputProps = BoxProps & RNTextInputProps & LocalInputProps;

const defaultProps = { labelTextColor: 'gray300', placeholderTextColor: 'gray300', variant: 'bordered' };

const useProps = createHook<InputProps>(
  (props) => {
    const ref = React.useRef();
    const boxProps = Box.useProps({ ...props, elementRef: mergeRefs(ref, props.elementRef) });

    const { theme } = useTheme();

    return { ...boxProps, placeholderTextColor: palette(props.placeholderTextColor)({ theme }) };
  },
  { defaultProps, themeKey: 'native.Input' }
);

export const Input =
  false && Platform.OS === 'web'
    ? WebInput
    : createComponent<InputProps>(
        (props) => {
          const { label, labelProps } = props;

          /////////////////////////////////////////////////////////////////////

          const inputRef = React.useRef();
          const [isFocused, setIsFocused] = React.useState(false);

          /////////////////////////////////////////////////////////////////////

          const handleBlur = React.useCallback(
            (e) => {
              setIsFocused(false);
              props.onBlur && props.onBlur(e);
            },
            [props]
          );

          const handleFocus = React.useCallback(
            (e) => {
              setIsFocused(true);
              // @ts-ignore
              inputRef.current && inputRef.current.focus();
              props.onFocus && props.onFocus(e);
            },
            [props]
          );

          /////////////////////////////////////////////////////////////////////

          const htmlProps = useProps(
            omitCSSProps({
              ...props,
              ref: inputRef,
              onBlur: handleBlur,
              onFocus: handleFocus,
              placeholder: !label ? props.placeholder : '',
            })
          );
          const element = createElement({
            children: props.children,
            component: styles.StyledInput,
            htmlProps: htmlProps,
          });

          /////////////////////////////////////////////////////////////////////

          const labelStartColor = usePalette(props.placeholderTextColor);
          const labelEndColor = usePalette(props.labelTextColor);

          const colorAnimation = React.useRef(new Animated.Value(0)).current;
          const topAnimation = React.useRef(new Animated.Value(12)).current;
          const fontSizeAnimation = React.useRef(new Animated.Value(16)).current;

          React.useEffect(() => {
            if (isFocused || inputRef.current?.value) {
              Animated.timing(colorAnimation, {
                toValue: 0,
                duration: 100,
                useNativeDriver: true,
              }).start();
              Animated.timing(topAnimation, {
                toValue: -8,
                duration: 100,
                useNativeDriver: true,
              }).start();
              Animated.timing(fontSizeAnimation, {
                toValue: 12,
                duration: 100,
                useNativeDriver: true,
              }).start();
            } else {
              Animated.timing(colorAnimation, {
                toValue: 0,
                duration: 100,
                useNativeDriver: true,
              }).start();
              Animated.timing(topAnimation, {
                toValue: 12,
                duration: 100,
                useNativeDriver: true,
              }).start();
              Animated.timing(fontSizeAnimation, {
                toValue: 16,
                duration: 100,
                useNativeDriver: true,
              }).start();
            }
          }, [colorAnimation, fontSizeAnimation, isFocused, topAnimation]);

          /////////////////////////////////////////////////////////////////////

          const color = colorAnimation.interpolate({
            inputRange: [0, 150],
            outputRange: [labelStartColor, labelEndColor],
          });

          return (
            <Box onClick={handleFocus} position="relative" {...pickCSSProps(props)}>
              {label && (
                <styles.StyledAnimatedLabel
                  color={color}
                  top={topAnimation}
                  fontSize={fontSizeAnimation}
                  variant={props.variant}
                  {...labelProps}
                >
                  {label}
                </styles.StyledAnimatedLabel>
              )}
              {element}
            </Box>
          );
        },
        {
          attach: {
            useProps,
            displayName: 'native.Input',
          },
          defaultProps,
          themeKey: 'native.Input',
        }
      );
