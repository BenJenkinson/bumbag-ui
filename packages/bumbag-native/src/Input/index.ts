import * as inputStyles from './Input.styles';

export * from './Input';
export { inputStyles };
