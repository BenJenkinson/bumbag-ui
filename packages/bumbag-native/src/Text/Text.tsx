import { TextProps as RNTextProps } from 'react-native';
import { createComponent, createElement, createHook } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import * as styles from './Text.styles';

export type LocalTextProps = {};
export type TextProps = BoxProps & LocalTextProps;

const useProps = createHook<TextProps>(
  (props) => {
    const boxProps = Box.useProps(props);
    return { ...boxProps };
  },
  { themeKey: 'Text' }
);

export const Text = createComponent<RNTextProps & TextProps>(
  (props) => {
    const textProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledText,
      htmlProps: textProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'Text',
    },
    themeKey: 'Text',
  }
);
