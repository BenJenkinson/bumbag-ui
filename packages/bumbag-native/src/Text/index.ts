import * as textStyles from './Text.styles';

export * from './Text';
export { textStyles };
