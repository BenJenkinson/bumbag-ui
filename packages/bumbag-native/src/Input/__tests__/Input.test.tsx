import * as React from 'react';
import { applyTheme } from 'bumbag/utils/applyTheme';
import { renderHook } from '@testing-library/react-hooks';
import { Input } from '../Input';
import { Text } from '../../Text';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Input />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Input color="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Input.styles.base should render correctly', () => {
    const { container } = render(<Input />, {
      // @ts-ignore REMOVE THIS
      theme: { Input: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Input.styles.base should render correctly', () => {
    const { container } = render(<Input />, {
      theme: {
        // @ts-ignore REMOVE THIS
        Input: { styles: { base: (props) => ({ backgroundColor: props.color }) } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Input
        overrides={{
          // @ts-ignore REMOVE ME
          Input: {
            styles: { base: { backgroundColor: 'red' } },
            defaultProps: {
              altitude: '100',
              padding: 'major-2',
            },
          },
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('applyTheme', () => {
  it('renders correctly', () => {
    const Card = applyTheme(Input, {
      styles: {
        base: {
          backgroundColor: 'red',
        },
      },
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<Input variant="test" />, {
      theme: {
        // @ts-ignore REMOVE THIS
        Input: {
          variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Input
        overrides={{
          // @ts-ignore REMOVE ME
          Input: {
            variants: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        variant="test"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Input, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      variants: {
        test: {
          defaultProps: {
            color: 'red',
          },
        },
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('modes', () => {
  it('should render correctly when colorMode is set globally', () => {
    const { container } = render(<Input />, {
      colorMode: 'test',
      theme: {
        // @ts-ignore DELETE ME
        Input: {
          modes: {
            test: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: { color: 'primaryTint' },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when colorMode set as prop', () => {
    const { container } = render(<Input colorMode="test" />, {
      theme: {
        // @ts-ignore DELETE ME
        Input: {
          modes: {
            test: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: { color: 'primaryTint' },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Input
        overrides={{
          // @ts-ignore REMOVE ME
          Input: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        colorMode="test"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Input, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      modes: {
        test: {
          styles: { base: { backgroundColor: 'red' } },
          defaultProps: { color: 'primaryTint' },
        },
      },
    });
    const { container } = render(<Card colorMode="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for color', () => {
    const { container } = render(<Input />, {
      // @ts-ignore REMOVE THIS
      theme: { Input: { defaultProps: { color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
