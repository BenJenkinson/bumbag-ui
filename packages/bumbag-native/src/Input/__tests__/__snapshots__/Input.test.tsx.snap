// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyTheme renders correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px; position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); font-size: 16px; height: 44px; padding: 16px 16px 16px 16px;"
    type="text"
  />
</div>
`;

exports[`defaultProps should render correctly for color 1`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(87, 79, 235); position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(252, 252, 253); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; color: rgb(87, 79, 235); font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`modes renders correctly for applyTheme 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(238, 237, 253); padding: 16px 16px 16px 16px; position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(238, 237, 253); font-size: 16px; height: 44px; padding: 16px 16px 16px 16px;"
    type="text"
  />
</div>
`;

exports[`modes renders correctly for overrides 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px; position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; box-shadow: 0px 1px 4px rgba(0,0,0,0.15); font-size: 16px; height: 44px; padding: 16px 16px 16px 16px;"
    type="text"
  />
</div>
`;

exports[`modes should render correctly when colorMode is set globally 1`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(238, 237, 253); position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; color: rgb(238, 237, 253); font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`modes should render correctly when colorMode set as prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(238, 237, 253); position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; color: rgb(238, 237, 253); font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`overrides should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px; position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; box-shadow: 0px 1px 4px rgba(0,0,0,0.15); font-size: 16px; height: 44px; padding: 16px 16px 16px 16px;"
    type="text"
  />
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(252, 252, 253); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(87, 79, 235); position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(252, 252, 253); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`theming Input.styles.base should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`theming Input.styles.base should render correctly 2`] = `
<div
  class="css-view-1dbjc4n"
  style="position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(252, 252, 253); border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-color: rgba(217,217,225,1.00); border-right-color: rgba(217,217,225,1.00); border-bottom-color: rgba(217,217,225,1.00); border-left-color: rgba(217,217,225,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-left-radius: 6px; border-top-right-radius: 6px; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; font-size: 16px; height: 44px; padding: 6.4px 12.8px 6.4px 12.8px;"
    type="text"
  />
</div>
`;

exports[`variants renders correctly for applyTheme 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(255, 0, 0); padding: 16px 16px 16px 16px; position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(252, 252, 253); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(255, 0, 0); font-size: 16px; height: 44px; padding: 16px 16px 16px 16px;"
    type="text"
  />
</div>
`;

exports[`variants renders correctly for overrides 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px; position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); font-size: 16px; height: 44px; padding: 16px 16px 16px 16px;"
    type="text"
  />
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="position: relative;"
>
  <input
    autocapitalize="sentences"
    autocomplete="on"
    autocorrect="on"
    class="css-textinput-11aywtz r-placeholderTextColor-fa7nb4"
    data-focusable="true"
    dir="auto"
    spellcheck="true"
    style="background-color: rgb(255, 0, 0); font-size: 16px; height: 44px;"
    type="text"
  />
</div>
`;
